pkg/spec/spec.go

```diff
@@ -397,6 +397,6 @@ func updateComponent(cfg GlobalCfg, field reflect.Value) error {
 }
 
 func skipUpdate(field reflect.Value) bool {
-	tp := field.Type().Name()
+	tp := reflect.TypeOf(field).Name()
```

Comment:
> why do we need to modify this?

That is because if there is pointer field in spec structs, is will panic when traversing because it will do `reflect.Type` on zero value.
The use of pointer fields (with ignore empty) in spec is to make code know the value is set by user or left unset.

Reply:
ok